cognite/client/data_classes/time_series.py

```diff
@@ -390,21 +493,31 @@ def data_set_id(self) -> _PrimitiveTimeSeriesUpdate:
         return TimeSeriesUpdate._PrimitiveTimeSeriesUpdate(self, "dataSetId")
 
     @classmethod
-    def _get_update_properties(cls) -> list[PropertySpec]:
-        return [
-            # External ID is nullable, but is used in the upsert logic and thus cannot be nulled out.
-            PropertySpec("external_id", is_nullable=False),
-            PropertySpec("name"),
-            # TimeSeries does not support setting metadata to an empty array.
-            PropertySpec("metadata", is_container=True, is_nullable=False),
-            PropertySpec("unit"),
-            PropertySpec("unit_external_id"),
-            PropertySpec("asset_id"),
-            PropertySpec("description"),
-            PropertySpec("is_step", is_nullable=False),
-            PropertySpec("security_categories", is_container=True),
-            PropertySpec("data_set_id"),
-        ]
+    def _get_update_properties(cls, item: CogniteResource | None = None) -> list[PropertySpec]:
+        if isinstance(item, (TimeSeries, TimeSeriesWrite)) and item.instance_id:
```

Comment:
Could we instead have a method that says `_get_update_properties_dms` (or something)? I think the caller of the function should decide which update properties to get rather than this method deciding for itself.